nncf/experimental/torch/fx/transformations.py

```diff
@@ -170,51 +169,63 @@ def insert_one_qdq_after_node(model: torch.fx.GraphModule, target_node: torch.fx
+    quantize_op_inputs[0] = feed_node
+    with ctx_manager(target_node):
         quantized_node = graph.create_node(node_type, quantize_op, tuple(quantize_op_inputs), {})
-        # use the same qparams from quantize op
+
+    # use the same qparams from quantize op
     dq_inputs = [quantized_node] + quantize_op_inputs[1:]
-    user_dq_nodes = []
-    with graph.inserting_after(quantized_node):
-        for user in target_node.users:
-            if user is quantized_node:
-                continue
-            user_dq_nodes.append((user, graph.call_function(dequantize_op, tuple(dq_inputs), {})))
+    if target_point.target_type == TargetType.OPERATOR_POST_HOOK:
+        user_dq_nodes = []
+        with graph.inserting_after(quantized_node):
+            for user in target_node.users:
+                if user is quantized_node:
+                    continue
+                user_dq_nodes.append((user, graph.call_function(dequantize_op, tuple(dq_inputs), {})))
+
+        for user, dq_node in user_dq_nodes:
+            user.replace_input_with(target_node, dq_node)
+    else:
+        with graph.inserting_after(quantized_node):
+            dq_node = graph.call_function(dequantize_op, tuple(dq_inputs), {})
 
-    for user, dq_node in user_dq_nodes:
-        user.replace_input_with(target_node, dq_node)
+        args = list(target_node.args)
+        args[target_point.input_port_id] = dq_node
+        target_node.args = tuple(args)
 
 
-def _insert_call_module(graph: torch.fx.Graph, target_node: torch.fx.Node, module_attr_name: str):
+def _insert_call_module(graph: torch.fx.Graph, target_point: PTTargetPoint, module_attr_name: str):
     """
     Inserts module call node to the graph after the target node.
 
     :param graph: Graph to insert module call node.
     :param target_node: Target node, module call node is being iserted just after the target node.
     :param module_attr_name: The name of the graph attribute which keeps the target module.
     """
-    with graph.inserting_after(target_node):
+    target_node, feed_node, ctx_manager = _get_data_for_insertion(graph, target_point)
+    with ctx_manager(target_node):
         return graph.create_node(
-            "call_module", module_attr_name, (target_node,), {}, name=module_attr_name + "_graph_node"
+            "call_module",
+            module_attr_name,
+            (feed_node,),
+            {},
+            name=f"{module_attr_name}_{str(target_point.target_type)}_graph_node",
         )
 
 
-def _get_target_node(graph: torch.fx.Graph, target_point: PTTargetPoint) -> torch.fx.Node:
-    """
-    Returns TorchFX graph node correspondent to the target point.
-
-    :param graph: Target torch.fx.Graph.
-    :param target_point: A target point to find the target node.
-    :return: TorchFX graph node correspondent to the target point.
-    """
-    # TODO(dlyakhov): Support node insertion on a specific input port id.
+def _get_data_for_insertion(
```

Comment:
Bad function , too general code. That should be spited to several short function this clean description.
Example:
- get_graph_node_by_name (already exists)
- get_fead_node
- get_insertion_point_ctx

Reply:
Done